master/internal/usergroup/api_groups.go

```diff
@@ -75,6 +86,16 @@ func (a *UserGroupAPIServer) GetGroups(ctx context.Context, req *apiv1.GetGroups
 		}
 	}
 
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	searchResults, err = AuthZProvider.Get().FilterGroupsList(*curUser, searchResults)
+	if err != nil {
+		return nil, err
+	}
+
 	return &apiv1.GetGroupsResponse{
 		Groups: searchResults,
 		Pagination: &apiv1.Pagination{
```

Comment:
I think this pagination is throw off by the filtering of the group list, like if we can only view 1 group out of a 100 we will get a total back of 100.

Can we make ```FilterGroupList``` apply a bun filter like ```GetExperiments``` does?

Reply:
ah good point